### PR TITLE
Acct users

### PR DESCRIPTION
**Pull request by gvonkoss**
test fixes pending

### CODE REVIEW

File: server/controllers/account.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const fetch = require('node-fetch');
+const url = 'https://api.myjson.com/bins/18azt3';
+const users = require('./users');
+
+module.exports = async (req, res, next) => {
+	try {
+		let response = await fetch(url);
```
Comment (laurieboyes) [inline server/controllers/account.js]: Plz can we use `const` for any variable we never reassign?

File: server/controllers/account.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		let account = await response.json();
+		let querystring = '?q=identities.user_id.raw: '; //because Lucene and auth0. whatever.
+
+		for (let id of account.user_ids) {
+			id = id.replace('auth0|', '');
+			querystring = querystring + id + ' OR ';
+		}
+
+		let model = await users.getAllUsers(account, querystring.slice(0,-4));
```
Comment (laurieboyes) [inline server/controllers/account.js]: Nice work on the query string fiddling logic 💯 Would be really good to write a test for it which tests the value of the params we pass to `getAllUsers`, for documentation if nothing else

File: server/controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+//Add user
+
+const addUser = async (body) => {
+	const newUser = {
+		email: body.email,
+		username: body.email,
+		password: body.password,
+		connection: 'Username-Password-Authentication',
+		verify_email: true,
+		nickname: `${body.firstName}.${body.lastName}`
```
Comment (laurieboyes) [inline server/controllers/users.js]: I reckon we should store these individually in the `given_name` and `family_name` fields (e.g. in the code sample for this the getUsers docs: https://auth0.com/docs/api/management/v2#!/Users/get_users)

File: server/lib/auth0-client/authenticate.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const fetch = require('node-fetch');
```
Comment (laurieboyes) [inline server/lib/auth0-client/authenticate.js]: Ah nice, I like that you've encapsulated this ✨. Could we rename the file to make it clear that it both does the authentication and makes the request? `authenticated-fetch` maybe?

File: server/lib/auth0-client/delete-user.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const authenticate = require('./authenticate');
+
+const user = async (id) => {
+	const url = `https://scoutasia.eu.auth0.com/api/v2/users/auth0|${id}`;
+	const opts = {
+		method: 'DELETE'
+	};
+
+	const res = await authenticate(url, opts);
+	console.log(res);
```
Comment (laurieboyes) [inline server/lib/auth0-client/delete-user.js]: Comment left in by accident?

File: server/lib/auth0-client/update-users.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const authenticate = require('./authenticate');
```
Comment (laurieboyes) [inline server/lib/auth0-client/update-users.js]: This is just updating one user right? Can we rename the file `update-user`?

File: views/create.html [hunk 7]
```diff
@@ -1,9 +1,5 @@
-
-
 <div class="umi__user">
-
-	<form class="umi__user--form" action="/save" method="post">
-		<input type="hidden" name="id" id="umi__user--id">
+	<form class="umi__user--form" action="/add" method="post">
```
Comment (laurieboyes) [inline views/create.html]: Ah I didn't spot this before. In BEM, the `--` is meant to indicate 'modifiers', so different versions of a component (http://getbem.com/naming/). I reckon we either want to use another `__` so we have like nested blocks: `umi__user__form` or have a flat block__element structure so make it `umi__user-form`.

_Or_ we could get crazy and think about swapping out BEM for functional CSS like in this dope ass blog Matt H tweeted. I've never used it, so I can't vouch for it personally, but it looks good on paper. Shrug. I'm happy to keep going as we are, but this is an interesting read anyway: https://adamwathan.me/css-utility-classes-and-separation-of-concerns/

File: server/controllers/account.js (hunk 2 above)
Comment (laurieboyes) [reply server/controllers/account.js]: Ah saying that, I've just remembered we'll be porting this code into the API 🤔 but we can totally port the tests too. We'll just see where it is when you get around to testing.

File: server/app.js [hunk 8]
```diff
@@ -1,52 +1,31 @@
...
-app.engine('html', handlebars.engine);
-app.set('view engine', '.html');
-
-app.use(express.static('public'));
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: true }));
-app.use(require('cookie-parser')());
-app.use(s3o);
-app.use(require('./middleware/allow-list-auth'));
+const app = require('./config');
```
[the remark this answers is not in the document]
Comment (gvonkoss) [reply server/app.js]: ✨

File: circle.yml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+machine:
+  node:
+    version: 7.10.0
```
[the remark this answers is not in the document]
Comment (gvonkoss) [reply circle.yml]: Because I copy pasted 🙈

File: server/controllers/account.js (hunk 2 above)
Comment (gvonkoss) [reply server/controllers/account.js]: I'll write _all of the  tests_ o/ shortly

**Pull request merged**